### ISSUE TITLE
Logo

### ISSUE THREAD

**Issue by benanne** [benanne/Lasagne#183]
We briefly discussed this in the dev meeting, but it would be great to have a logo by the time we put out the first release :) There were various suggestions in the email conversation afterwards, including getting one made through https://www.fiverr.com/ . Is that still our best option?

I'd like to have this ready by the time we release Lasagne 0.1, so we can use it in our docs and release announcements.

**Comment by ebattenberg** [comment]
I'd be willing to throw in more than $5 to get a little more thought put into it if there's someone we could hire to make one.  But fiverr sounds like a cool option.

**Comment by f0k** [comment]
We can still use fiverr as a starting point if we feel we need a more refined logo. But it seems you'd get something useful out of it, and we could try to keep in contact with whoever does the logo to do additional logos in the same style for side projects or graphics for our web site, for more than $5, to fight these dumping prices a bit.

What we still need is some textual description of what we want. I assume that the logo should depict a lasagna. Some questions we should answer:

1. Which perspective? From the side, so you clearly see the layers, or isometric, so you see the layers less prominently, but you see that it's a stack of two-dimensional things? Or true perspective?
2. Which style? Abstract with clear lines and corners, or cartoonish?
3. Which color? Black & white, 3-4 colors, many colors?
4. What features should be present? Bolognese, Bechamel, Pasta, Cheese? Structured crust? Sauce dripping out? Dishes or not?

We could also commission multiple variants and decide afterwards. I'd be fine sharing costs as well.

**Comment by benanne** [comment]
I came into some money recently, in no small part thanks to Lasagne, so I'd be okay with covering the cost by myself ;)

Is it actually possible at all to pay more than $5 on fiverr.com? The name would imply otherwise :) I agree that $5 for a decent logo is ridiculously little.

**Comment by craffel** [comment]
> Is it actually possible at all to pay more than $5 on fiverr.com? The name would imply otherwise :) I agree that $5 for a decent logo is ridiculously little.

The way it works is that usually $5 is for the most barebones thing - e.g., for a logo, it will be a medium-sized jpg.  If you want a vectorized version, rush service, handmade fonts, sometimes even color, etc., there's an upcharge.  It still usually ends up being cheap.  Crucial Python ended up paying $5 for this:

![image](https://cloud.githubusercontent.com/assets/417568/6922580/21dd93ba-d79a-11e4-9c76-0a6faded20e0.png)

Maybe all interested people should either buy fiverr logos or design their own based on their own preferences wrt @f0k's questions https://github.com/benanne/Lasagne/issues/183#issuecomment-88129079 and then we can vote.

**Comment by MartinThoma** [comment]
What do we want as a logo? Here are my guesses (without having experience with logo design):

* The logo should contain a lasagne
* The lasagne should look delicious
* It should probably be a clipart image, not a photograph (because photographs don't look nice when they  are scaled) and not line art (they look a bit "sketchy")
* It should look nice in 100x100 pixel (GitHub organization image size) and be recognizable in 20x20 pixels (not necessarily the same image, eventually the 20x20 could only be a 'L' in the same color scheme?)
* The format should be png or svg

Here are some examples of lasagnes:

* http://www.istockphoto.com/vector/lasagna-10735610?st=893df70 : lasagne on fork
* http://www.istockphoto.com/vector/lasagna-29263600?st=893df70 : lasagne with parmesan and basil
* http://www.istockphoto.com/vector/lasagna-50620356?st=893df70 : lasagne with basil and text
* http://www.istockphoto.com/vector/baked-lasagna-59341212?st=893df70 : lasagne with basil and tomato
* http://www.vectorstock.com/royalty-free-vector/detailed-icon-lasagna-vector-2796135 : lasagne with basil and tomato

I think a "pure" lasagne would be a good image. This would leave room for follow-up projects which extend Lasagne. They could, like @ebattenberg proposed, be called "parmesan", "basil", "fork", "tomato", ...

[Logoturnament](http://logotournament.com/) is a site I always wanted to try, but $275 is too expensive for me. I would contribute up to $10 for a logo, not more.
(Caution with fiverr.com. I took a look at some proposals (or whatever they are called) and found one note that they make only sketches for $5. I also didn't find the format in which we get the logo.)

**Comment by benanne** [comment]
I think all of those examples are a bit too 'realistic' and detailed to work as a logo, to be honest - so I don't think clipart is going to be a good source to base this off. Something more stylized would be better suited.

$ 275 seems a bit excessive for now :p I'm all for people receiving fair rewards for their work, but if we can't get it done professionally for cheap we should probably just whip something up ourselves.

**Comment by f0k** [comment]
I think a less detailed / more cartoonish version of the third one could work. Or a very abstract lasagna that just consists of a quad and some lines.

**Comment by MartinThoma** [comment]
> $ 275 seems a bit excessive for now :p I'm all for people receiving fair rewards for their work, but if we can't get it done professionally for cheap we should probably just whip something up ourselves.

The best I can come up with is http://pastebin.com/PikSC9us ... I'm definitely not an artist :-/

![Rendered lasagne ... ](http://i.imgur.com/x8ybvUE.png)

**Comment by fchollet** [comment]
You guys are making me hungry. But couldn't you train a DCNN + RNN combination (something in the spirit of DRAW [1]) on a large number of logos, then seed it with a picture of lasagnes to get it to generate a lasagnes-inspired logo? 

The result would most likely be interesting. It would also cost a lot __more__ than $5. 

[1] http://arxiv.org/pdf/1502.04623.pdf

**Comment by f0k** [comment]
@fchollet: Well, feel free to try it and post your result ;) @skaee has a [Lasagne-based DRAW implementation](https://github.com/skaae/lasagne-draw). (By the way, it seems you're doing something very similar to Lasagne called keras, also under MIT license? We might exchange some code.)

@MartinThoma: I think the uppermost vertex should be further to the left and the result could be confused with a sandwhich, but it's a start! :)

**Comment by ebenolson** [comment]
Has anyone trained DRAW yet to generate letters? An animation of a network writing the name would be pretty neat.

I decided to give fiverr a try, this is what $5.50 got me: ![Imgur](http://i.imgur.com/gw9G6sr.jpg)

Hope I don't offend anyone who's offered suggestions so far, but I really think you should spend a decent amount on a professional if you're going to do this. Good design is hard, and in my opinion anything but an outstanding logo is worse than not having one.

If you do decide to go that route, I'd be willing to make a financial contribution too.

**Comment by skaae** [comment]
Do you know a character dataset like mnist?

**Comment by MartinThoma** [comment]
@skaae Are you asking because you want to generate Lasagne?

I have created write-math.com, where you can classify single symbols. I don't have much data for Latin letters, but at least 50 recordings for each character. You can download it here:
http://write-math.com/data
but it is on-line data. So you have to render it first.

**Comment by skaae** [comment]
dunno :) Just asking, I thought about training it on latin letter, but i don’t know any large datasets.
50 examples pr class is too small.

**Comment by MartinThoma** [comment]
How much do you need per class? (I am currently talking with creators of
other databases. I guess I could get up to 150 examples per class in a
couple of days)

**Comment by skaae** [comment]
I dont know, quite a a lot I guess. MNIST is ~60K, i guess something in that ballpark?

maybe we should have this discussion in an non-issue thread?

**Comment by benanne** [comment]
> maybe we should have this discussion in an non-issue thread?

That would be appreciated :)

**Comment by fchollet** [comment]
@skaae, @MartinThoma : it sounds like you are looking for something like: http://www.cs.toronto.edu/~graves/handwriting.html

**Comment by f0k** [comment]
> I decided to give fiverr a try, this is what $5.50 got me:

Thanks a lot for trying it! Unfortunately, that one is just a slightly rotated free clipart easy to find via google image search: http://de.clipartlogo.com/image/lasagna-clip-art_369009.html
We can't really blame anyone for doing that, so maybe you're right and we should try another place. However, @craffel's logo looks like someone actually drew it on request (I mean, it's probably not easy to find a guitar-playing Python clipart, compared to finding a lasagne).

> I really think you should spend a decent amount on a professional if you're going to do this.

Or find a friend to do it for a few beers. I'll investigate if I know somebody who can draw even better than Martin :)

**Comment by f0k** [comment]
What about a really minimalistic stylized one, with a lot of layers?

![lasagne8](https://cloud.githubusercontent.com/assets/629706/8236556/6789c2b6-15e9-11e5-81f9-8e233c41fc93.png)

Did it in Inkscape, it's fully vectorized. If you like it in principle, I'd gradually add more layers. I'd like to end up with something that has more layers than a usual lasagna, but is still recognizable.

**Comment by moscow25** [comment]
Awesome!

Probably too many layers already at logo-scale... perfect for a simple,
iconic look.

Best,
Nikolai

**Comment by ebenolson** [comment]
How about squashing it a bit and stacking another at 90 degrees to make an `L`?

**Comment by f0k** [comment]
> Probably too many layers already at logo-scale...

Hmm, yes, maybe we should leave it at that. It's already higher than a usual lasagna.

> How about squashing it a bit and stacking another at 90 degrees to make an `L`?

Creative idea, but I don't think it would look good. I've changed my post to link to the Inkscape SVG file instead, feel free to have a go!

**Comment by ebenolson** [comment]
(terrible sketch just for illustration)
![drawing](https://cloud.githubusercontent.com/assets/544718/8237969/dbce1c54-15c0-11e5-9c2f-3ee74ab8dfb8.png)

**Comment by benanne** [comment]
> Hmm, yes, maybe we should leave it at that. It's already higher than a usual lasagna. But let's wait what others think about the style at all.

Looks good, not sure about the wavy lines. It might look better with straight lines (or wavy lines with a lower frequency).

**Comment by MartinThoma** [comment]
@f0k Looks nice :-) However, I really would like some color in it.
@ebenolson Why are there some lines from top to down?

**Comment by craffel** [comment]
I was hoping there would be a tie-in between the wiggly lasagne layers and some sort of traditional graphical representation of neural networks.  Something like the following sketch (made in `matplotlib`, because what else would I use?).  If @f0k worked this into his design I'd be very behind it.

![unknown](https://cloud.githubusercontent.com/assets/417568/8239831/fb2f1cd0-15c3-11e5-813d-d116e336d08a.png)

**Comment by benanne** [comment]
I like the idea, but that has the potential to look very messy when the logo is quite small. Imagine it in the top left corner on our docs page for example: http://lasagne.readthedocs.org/en/latest/
It should still look good and be recognizable in that size.

**Comment by craffel** [comment]
Hm, yes, I'm not sure how it would look that small...

**Comment by f0k** [comment]
> (terrible sketch just for illustration)

Looks fun! But I think it's not necessary to form an L. I like the drawing style, though, and having the top layer in a wider stroke. I guess we'll be able to come up with something for a logo on our own!

> It might look better with straight lines (or wavy lines with a lower frequency).

I had straight lines before, and also played with different distances to form different layer types. Straight lines don't really look like lasagne, though. Go ahead and try yourself if you've got a few minutes -- just open it up in Inkscape and disable or modify the path effect of the top two layers (all the others are clones). You'll find that somewhere in the "Path" menu.

> However, I really would like some color in it.

Try it! I like how the lasagne is open on the sides now, but maybe colored layers would look good as well.

> If @f0k worked this into his design I'd be very behind it.

Feel free to try yourself! Your sketch looks interesting, but reminds me of a bicycle chain more than a lasagna. It might be possible to just add some lines to my sketch that symbolize the connections between layers, but as Sander said, this could become messy.

**Comment by benanne** [comment]
I'm not enough of an Inkscape wizard to get the path effects to look decent, I guess :p every time I try to change something it ends up looking terrible. Maybe they should not be straight as I said, but a slightly lower frequency (and maybe higher amplitude) could look better. A smaller slice of Lasagne, basically. The current version doesn't scale down very well, the lines just start to look badly drawn instead of wavy.

**Comment by f0k** [comment]
> I'm not enough of an Inkscape wizard to get the path effects to look decent, I guess :p every time I try to change something it ends up looking terrible.

Not an expert either. I didn't know about path effects until yesterday. I don't know if there's an easy way to change the frequency -- the amplitude can be set directly, but for the frequency you may have to change the effect. You can copy out the path from the path effect (there's an extra button in the path effect settings, near "Pattern source"), paste it into the document, stretch it, copy it and paste it into the path effects again.

Some more variants:
<img src="http://ofai.at/~jan.schlueter/img/lasagne9.svg" width="200"/> <img src="http://ofai.at/~jan.schlueter/img/lasagne12.svg" width="200"/> <img src="http://ofai.at/~jan.schlueter/img/lasagne10.svg" width="200"/>
<img src="http://ofai.at/~jan.schlueter/img/lasagne11.svg" width="200"/> 

These use Eben's style of having the top layer in a stronger stroke (and I also increased the stroke width for the hidden layers). The colored one has an artifact, but it was just for illustration. I think it doesn't work well, at least not in this way. Martin, feel free to come up with something better if you want colors, but I'm a bit skeptical!

**Comment by MartinThoma** [comment]
@f0k I like the colored one :-) Why is there a white line on the top?

**Comment by f0k** [comment]
>> The colored one has an artifact because the top layer is composed of two cloned paths rather than being a single rectangle, but it was just for illustration.

> @f0k I like the colored one :-) Why is there a white line on the top?

Because the top layer is composed of two cloned paths rather than being a single rectangle. (Moving the top path down a bit should close the gap without increasing the file size as joining the paths would, but I didn't bother because I prefer the uncolored ones anyway.)

**Comment by benanne** [comment]
I tried experimenting with thicker lines / less waviness so it would look better when downscaled, but I'm having a hard time making it look good. Overall I think straight lines is better. It may look less inherently "Lasagne-like" but it will be juxtaposed with the library name almost all the time so people will get the idea :) And I think it will look better when it's small. And simpler is better, imo.

**Comment by MartinThoma** [comment]
@f0k Ooops, sorry for not reading the text.

I just quick-fixed the artifact issue: https://gist.github.com/MartinThoma/c9fe04c4a37e8e92bd02 (by the way, where did you upload the images?)

**Comment by ebenolson** [comment]
Just curious, where is this logo intended to be used? 

@MartinThoma you mentioned the github organization image is 100x100, what is 20x20 for?

**Comment by MartinThoma** [comment]
@ebenolson I'm sorry, I don't know why I wrote 20x20. Either 16x16 or 32x32 (see http://stackoverflow.com/q/4014823/562769) are used for favicons (and could be used in the Lasagne documentation). But 20x20 doesn't make any sense to me now.

**Comment by ebenolson** [comment]
@MartinThoma if you save as a png you can drag and drop into the comment box.

Just playing around with your version a bit:
![lasagne-logo](https://cloud.githubusercontent.com/assets/544718/8278966/e73ffd0c-189d-11e5-8417-410eadc17afa.png)

**Comment by benanne** [comment]
> Just curious, where is this logo intended to be used?

Everywhere it is appropriate I guess? :) The documentation, the mailing list, people's slide decks, ...

But as I just posted on the mailing list, this isn't a show stopper for the first release as far as I'm concerned. We can always add it in later, and I have a feeling this discussion may rage on for some time anyway!

**Comment by f0k** [comment]
> Overall I think straight lines is better.

I prefer wiggly lines because they look like lasagne even without the name spelt out next to it.

> And I think it will look better when it's small.

Which size? It's common to have different versions for different scales, i.e., the favicon wouldn't be just a scaled-down version of the large one, but re-drawn to show the same key elements (wiggly lines, but maybe fewer layers).

> I have a feeling this discussion may rage on for some time 

Seem so :) But I think we're on a much better route than using fiverr now. It seems we're capable of designing a logo on our own, and we've got a lot more control this way.

> by the way, where did you upload the images?

Just to my website.

> if you save as a png you can drag and drop into the comment box.

I think it's a bug that you can't drag and drop SVG files, since the ones I uploaded to my website got converted into *.githubusercontent.com links just like your PNGs. By the way, when using HTML syntax, you can also downscale them for display: `<img src="..." width="200"/>`

> Just playing around with your version a bit

I still prefer the monochrome one. The colored ones are too busy, and the monochrome ones can be easily adapted to any color scheme.

**Comment by benanne** [comment]
> I prefer wiggly lines because they look like lasagne even without the name spelt out next to it.

But as I said, it's rarely going to be used in a context where the name is not mentioned.

> Which size? It's common to have different versions for different scales, i.e., the favicon wouldn't be just a scaled-down version of the large one, but re-drawn to show the same key elements (wiggly lines, but maybe fewer layers).

That's also an option of course. Probably better not to change the number of layers though, that'll look too different.

> It seems we're capable of designing a logo on our own, and we've got a lot more control this way.

I don't think we are quite there yet. We have some promising mockups but we're a way away from having a professional looking logo, at the moment :p

> I still prefer the monochrome one. The colored ones are too busy, and the monochrome ones can be easily adapted to any color scheme.

Yeah, I would stick to monochrome as well. Maybe we should give it a default colour though, black is a little boring :)

**Comment by stonebig** [comment]
impy-lasagne ?
![impylasagne](https://cloud.githubusercontent.com/assets/4312421/8339939/793faeb8-1abb-11e5-83c4-22ec20fd955d.png)

**Comment by eickenberg** [comment]
no snakes?

![lasagne_logo](https://cloud.githubusercontent.com/assets/1306635/8340677/56c70976-1ac0-11e5-9b79-5764c7546241.png)

(better, but outside my circle of comptence, would be to have the snake sidling through the layers)

**Comment by fchollet** [comment]
Love the snake. Genius! 

> have the snake sidling through the layers

[YES](http://digitalartanddesign.org/assets/photoshop/images/animated-snake-example2.gif)

**Comment by wuaalb** [comment]
Had a go at it as well.. 
![lasagne_test1](https://cloud.githubusercontent.com/assets/8669162/10200883/4bdb3d30-67a9-11e5-9ae5-c0f91c2a7f69.png)
![lasagne_test1_200x200](https://cloud.githubusercontent.com/assets/8669162/10200888/50991ea0-67a9-11e5-8af8-d143201b511e.png)

**Comment by fchollet** [comment]
If you removed the bottom snake and turned each lasagne layer into a stylized snake (with alternating orientation?) it would look pretty interesting and original, I think...

**Comment by ebattenberg** [comment]
At work, we did a bunch of polls to narrow down prototype designs we got
from 99designs.  That was pretty productive.  We got to see a lot of
designs.  No idea how much it costs.  I think you just make a bid and then
a bunch of designers give you their try out designs.

**Comment by f0k** [comment]
Thanks for your contribution, nice idea with the snake-plate! And I still like black-on-white wobbly lines for stylizing Lasagne. But I don't think we need a snake in there just because it's Python... Not sure why @fchollet is so keen on snakes in Lasagne when he doesn't have any in his library's logo either :) http://keras.io/img/keras-logo-small.jpg

> At work, we did a bunch of polls to narrow down prototype designs we got from 99designs. That was pretty productive. [...] No idea how much it costs.

Starts at 249 EUR: http://en.99designs.at/pricing
That's still cheaper than hiring a designer, but probably you'll also get worse results. Which is not to say that I expect better results from doing the logo ourselves :) And the concept doesn't sound too bad.
